Python/fhirreference.py

```diff
@@ -73,4 +78,20 @@ def processedReferenceIdentifier(self):
         
         # TODO: distinguish absolute (has "://") and relative URLs
         return None
-    
+
+    def _validate_element_value(self, element, value):
+        """ Validates value of the element.
+        """
+        if element == 'reference':
+            result = urlparse(value)
+            if result.scheme in ('https', 'http'):
+                if result.netloc == '':
+                    return False
+            elif result.scheme != '':
+                return False
+
+            if self.reference_regexp.fullmatch(result.path) is None:
```

Comment:
If make regexp like `'(.+\/)?[A-Z][a-zA-Z]+\/.{1,64}(\/_history\/.{1,64})?$'` (with final anchor) simple `reference_regexp.match(result.path)` could be used.